### PR TITLE
Fix documentation based on dartdoc's warnings

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -94,8 +100,12 @@ dependencies:
     '--favicon=favicon.ico',
     '--use-categories',
     '--category-order', 'flutter,Dart Core,flutter_test,flutter_driver',
+    '--show-warnings',
+    '--auto-include-dependencies',
```
Comment (Hixie) [inline dev/tools/dartdoc.dart]: fyi @jcollins-g i'm planning on turning on --auto-include-dependencies

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 2]
```diff
@@ -41,7 +41,7 @@ class CupertinoNavigationBar extends StatelessWidget implements PreferredSizeWid
     this.trailing,
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: cc @xster this is the change i was talking about. (part 1/2)

File: packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart [hunk 3]
```diff
@@ -569,12 +569,12 @@ class _NoDefaultValue {
   const _NoDefaultValue();
 }
 
-/// Marker object indicating that a DiagnosticNode has no default value.
```
Comment (Hixie) [inline packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart]: cc @jacob314 there are a number of changes relating to your recent patch throughout this PR, starting here

File: packages/flutter/lib/src/material/tabs.dart [hunk 4]
```diff
@@ -458,10 +458,13 @@ class TabBar extends StatefulWidget implements PreferredSizeWidget {
   final double indicatorWeight;
 
   /// The horizontal padding for the line that appears below the selected tab.
-  /// For [isScrollable] tab bars, specifying [kDefaultTabLabelPadding] will align
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tabs.dart]: cc @HansMuller this is the change we talked about earlier

File: packages/flutter/test/cupertino/nav_bar_test.dart [hunk 5]
```diff
@@ -67,4 +69,44 @@ void main() {
     );
     expect(find.byType(BackdropFilter), findsOneWidget);
   });
+
+  testWidgets('Verify styles of each slot', (WidgetTester tester) async {
```
Comment (Hixie) [inline packages/flutter/test/cupertino/nav_bar_test.dart]: cc @xster this is the test for the change above. (part 2/2)

File: packages/flutter_driver/lib/flutter_driver.dart [hunk 6]
```diff
@@ -25,8 +25,7 @@ export 'src/error.dart' show
   LogRecord,
   flutterDriverLog;
 export 'src/find.dart' show
-  SerializableFinder,
```
Comment (Hixie) [inline packages/flutter_driver/lib/flutter_driver.dart]: cc @yjbanov this and the next change are to driver's docs

File: packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart (hunk 3 above)
Comment (jacob314) [inline packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart]: LGTM for all the tree_diagnostics_mixin changes.
Thanks for cleaning this up!

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (devoncarew) [inline dev/tools/dartdoc.dart]: Just so you know, she's OOO through part of next week.

File: packages/flutter_driver/lib/flutter_driver.dart (hunk 6 above)
Comment (yjbanov) [inline packages/flutter_driver/lib/flutter_driver.dart]: LGTM

File: dev/tools/dartdoc.dart [hunk 7]
```diff
@@ -106,8 +116,18 @@ dependencies:
     args,
     workingDirectory: 'dev/docs',
   );
-  printStream(process.stdout);
-  printStream(process.stderr);
+  printStream(process.stdout, prefix: 'dartdoc:stdout: ',
+    filter: kVerbose ? const <Pattern>[] : <Pattern>[
+      new RegExp(r'^generating docs for library '), // unnecessary verbosity
+      new RegExp(r'^pars'), // unnecessary verbosity
```
Comment (yjbanov) [inline dev/tools/dartdoc.dart]: If this is filtering out "parse" and "parsing", then maybe `r'^pars(e|ing)'`?

**Review by yjbanov — APPROVED** [review]
LGTM

File: dev/tools/dartdoc.dart (hunk 7 above)
Comment (Hixie) [inline dev/tools/dartdoc.dart]: i'm tempted to just remove these two. it was only to make it a little easier while i was going through the output.

**Comment by Hixie** [comment]
Yegor was your LGTM for the whole patch or just the driver part?

File: packages/flutter/lib/src/cupertino/nav_bar.dart (hunk 2 above)
Comment (xster) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: thanks!

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 8]
```diff
@@ -77,15 +78,32 @@ class CupertinoNavigationBar extends StatelessWidget implements PreferredSizeWid
...
+    );
+
+    final Widget styledTrailing = trailing == null ? null : DefaultTextStyle.merge(
+      style: actionsStyle,
+      child: trailing,
+    );
+
+    // Let the middle be black rather than `actionsForegroundColor` in case
+    // it's a plain text title.
+    final Widget styledMiddle = middle == null ? middle : DefaultTextStyle.merge(
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: why not `middle == null ? null` like you did above?

File: packages/flutter/lib/src/cupertino/nav_bar.dart [hunk 9]
```diff
@@ -77,15 +78,32 @@ class CupertinoNavigationBar extends StatelessWidget implements PreferredSizeWid
...
+      style: const TextStyle(
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: what about just actionsStyle.copyWith so everything except the color is in one place?

**Review by xster — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/cupertino/nav_bar.dart (hunk 8 above)
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: copypasta. i'll fix it.

File: packages/flutter/lib/src/cupertino/nav_bar.dart (hunk 9 above)
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: ok

Comment (Hixie) [inline packages/flutter/lib/src/cupertino/nav_bar.dart]: done

**Comment by Hixie** [comment]
Ok I think the three LGTMs above cover all the code here so I'm going to land on green.

**Comment by yjbanov** [comment]
> Yegor was your LGTM for the whole patch or just the driver part?

LGTM for the driver parts, and somewhat RSLGTM for others.

**Pull request merged**